src/ObjectStateMutations.js

```diff
@@ -154,14 +154,24 @@ export function estimateAttributes(
   return data;
 }
 
+function nestedSet(obj, key, value) {
+  const path = key.split('.');
+  for (let i = 0; i < path.length - 1; i++) {
+    if (!(path[i] in obj)) obj[path[i]] = {};
+    obj = obj[path[i]];
+  }
+  if (typeof value === 'undefined') delete obj[path[path.length - 1]];
+  else obj[path[path.length - 1]] = value;
```

Comment:
Can you wrap the if / else with brackets `{}`.

Reply:
@dplewis Done